python/pyarrow/tests/test_parquet.py

```diff
@@ -641,6 +641,36 @@ def make_sample_file(table_or_df):
     return pq.ParquetFile(buf)
 
 
+def test_byte_stream_split():
+    # This is only a smoke test.
+    arr_float = pa.array(list(map(float, range(100))))
+    data_float = [arr_float, arr_float]
+    table = pa.Table.from_arrays(data_float, names=['a', 'b'])
+
+    # Check with byte_stream_split for both columns.
+    _check_roundtrip(table, expected=table, compression="gzip",
+                     use_dictionary=False, use_byte_stream_split=True)
+
+    # Check with byte_stream_split for column 'b' and dictionary
+    # for column 'a'.
+    _check_roundtrip(table, expected=table, compression="gzip",
+                     use_dictionary=['a'],
+                     use_byte_stream_split=['b'])
+
+    # Check with a collision for both columns.
+    _check_roundtrip(table, expected=table, compression="gzip",
+                     use_dictionary=['a', 'b'],
+                     use_byte_stream_split=['a', 'b'])
+
+    # Try to use the wrong data type with the byte_stream_split encoding.
+    # This should throw an exception.
+    arr_int = pa.array(list(map(int, range(10))))
+    table = pa.Table.from_arrays([arr_int], names=['tmp'])
+    with pytest.raises(IOError):
+        _check_roundtrip(table, expected=table, use_byte_stream_split=True,
```

Comment:
Wouldn't it be more convenient for `use_byte_stream_split=True` to only apply to floating point columns? That way, if you have a table with e.g. 30 float columns and a bool columns, you don't have to enumerate the 30 float columns explicitly...

The error should only be raised if you explicitly pass a column name that's not floating point.

IOW, would be nice to have a test with mixed column types.

Reply:
I think it's not a big deal in the case of python. I also think that the dictionary and plain encodings work similarly.
Dictionary is enabled by default, but if one wants explicitly to use the plain encoding for some columns and dictionary for other ones, then one still has to specify a subset of the columns which would use one encoding or the other.
I think it's better to explicitly select encodings which only make sense in specific scenarios. The byte_stream_split encodings only makes sense for high-entropy floating point data, and likely for integer data. The latter has to be checked.